docs/discovery/provisioning.md

```diff
@@ -0,0 +1,456 @@
+    // SetMaxItems - Set the maximum number of items in the array property
+    SetMaxItems(max uint) ArrayPropertyBuilder
+}
+```
+
+Example of an array property allowing integer inputs with Min and Max constraints.
+
+```go
+apic.NewSubscriptionSchemaPropertyBuilder().
+  SetName("Array property").
+  SetDescription("Description of the Array property.").
+  IsArray().
+    AddItem(apic.NewSubscriptionSchemaPropertyBuilder().
+      SetName("HTTP Code").
+      IsInteger().
+        SetMinValue(100).
+        SetMaxValue(599))
+```
+
+Example of an array property allowing the user to choose between 2 kind of authorization:
+
+```go
+apic.NewSubscriptionSchemaPropertyBuilder().
+  SetName("Authorization").
+  SetDescription("Determines the type of authorization used.").
+  IsArray().
+  SetMaxItems(1).
+    AddItem(apic.NewSubscriptionSchemaPropertyBuilder().
+      SetName("OAuth2").
+      IsObject().
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("OAuth2 client id").
+          SetDescription("OAuth2 client id.").
+          SetRequired().
+          IsString()).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("OAuth2 client secret").
+          SetDescription("OAuth2 client secret.").
+          SetRequired().
+          IsString()).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("OAuth2 provider").
+          SetDescription("URL provider of access tokens. Example: http://mysite.com/oauth/token").
+          SetRequired().
+          IsString()).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("OAuth2 mode").
+          SetDescription("Whether to send client credentials in body or via header of the request.").
+          SetRequired().
+          IsString().
+          SetEnumValues([]string{"header", "body"})).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("OAuth2 scope").
+          SetDescription("Scope supported for the client credentials OAuth2 authorization. Example: READ").
+          IsString())).
+    AddItem(apic.NewSubscriptionSchemaPropertyBuilder().
+      SetName("Credentials").
+      IsObject().
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("Username").
+          SetDescription("Your username credential.").
+          SetRequired().
+          IsString()).
+        AddProperty(apic.NewSubscriptionSchemaPropertyBuilder().
+          SetName("Password ").
+          SetDescription("Your password credential.").
+          SetRequired().
+          IsString()))
+```
+
+## Handle Provisioning Requests
+
+The Agents SDK provides a registration method for the agent to implement an interface that is called when certain provisioning events happen in Amplify Central. Once registered the Agents SDK will watch for these events and trigger the appropriate method for the agent o handle that provisioning request.
```

Comment:
let's chop this one up and add a comma
Once registered, the Agents SDK will watch for these

You're missing a t for "to" in .. agent to handle
trigger the appropriate method for the agent to handle that provisioning request.

Reply:
chop up as in 2 paragraphs?